### ISSUE TITLE
JSON table error

### ISSUE THREAD

**Issue by pyluyten** [jerry40/the-old-mode#2]
i cannot make the-old func work. i have : "contacting host... JSON readtable error"
otoh i did not find where to put login, password...

can i submit something useful?

**Comment by jerry40** [comment]
Hi! You need to put these lines into your .emacs:
```
;; the-old-reader mode
(setq load-path (cons "~/path to the-old.el folder/" load-path))
(setq the-old-api-login "mail@gmail.com")  ;; your login
;(setq the-old-api-passwd "password")
;(setq the-old-api-token "oath_token")
(require 'the-old)
```
Note 2 lines are commented since it probably insecure to keep passwords and oauth token in the config file. Later I want to read them from environment values, not from the config.
When these lins are commented, a password will be asked at the start (`alt+x the-old`)
Also, to work with the old reader with this emacs mode you need to set password on https://theoldreader.com/ in settings and use it for authentication.
You can read details there https://github.com/theoldreader/api .
Let me know if it helps please! Also I'd appreciate your help in developing this mode. Let me think what parts can be improved first. 

Thanks!

**Comment by pyluyten** [comment]
Thanks

about load-path , yes i had it. But never setq load-path, instead (add-to-list 'load-path "...")
it would be good to have a message saying a login is necessary.

adding login, password and token, it was possible to see the list of messages. i have also tested marking read, staring an item. It's super cool to have the old reader in Emacs! and i'm sure many emacser will think the same.

what i missed first is "h" did not work. So, I had to look at your code to find keys. In the best world "h" behaves like magit, showing all possible keys.
then i missed filtering, at least a way to see all unread articles. Maybe it does exist? sometimes i did not understood the lambda in the code. maybe you want a welcome screen like mu4e to tell you "to see al unread, do this / to view subscriptions do this / add subscription do this...." (a bit like the "m" key actually?)

Then i noticed html rendering is not read only. 

- I suggest keeping this bug open because of the lack of feedback when login is missing
- i don't know yet if i can help regarding code. maybe i can drop a .texi to build some doc. Or i can open bugs to keep track of missing pieces.

**Comment by jerry40** [comment]
Yes, UI isn't very usable. Oh I commented 'h' out! Don't remember why actually:

`  ;; help
  ;(define-key the-old-menu-mode-map "h" 'menu-quick-help)`

If you have ideas how to make UI better, please describe them.
Filtering does not work, I think it is the first what is needed to be done.
Doc or short description of the mode would be just great! As you can see, my English isn't very good for writing docs. Let's start it. I wiil participate in writing docs with my pleasure!
Welcome screen can be annoying, what do you think? Users may want to get into news list quickly. Perhaps we can add a keybinding to show simple documentation in another buffer.

**Comment by pyluyten** [comment]
> Welcome screen can be annoying, what do you think? Users may want to get into news list quickly. Perhaps we can add a keybinding to show simple documentation in another buffer.> 

yes sure this makes sense. When opening "the-old", user sees a list of subscriptions, presses "h" to see how to filter/view/other. So it's a quick interface but offering help.

I am starting a manual so i will note what i see. Attached pdf comes from a .texi, the texinfo GNU format. But the manual will depend on issue #4 

[the-old.pdf](https://github.com/jerry40/the-old-mode/files/1442505/the-old.pdf)

**Comment by jerry40** [comment]
Wow, the document looks promising! Issue #4 fixed , check it please if you can.

**Comment by pyluyten** [comment]
I have amended the documentation given new keybindings. I did push it, so i hope it is alright for you. And since ongoing work is on "filters" branch i did push to this branch. Once again i hope this is fine. I can revert commit if any issue.

You can look at the pdf. Or, if you open the the-old.texi file, run command 'makeinfo-buffer to generate an Info file, which is just another way to visualize the doc.

**Comment by jerry40** [comment]
Looking good, let me add myself to authors macros and we'll proceed (or do authors contain only PDF author? Correct me if I've got it wrong please). Also, I think 1-4 keys are very useful, for me it is interesting to use them!

**Comment by pyluyten** [comment]
Authors "fixed" (well, if you want "author = jerry40", without any information about name or email). Otherwise you might edit macros.texi

yes i want to add filtering to doc. Can you describe what is the roadmap for the filtering part? so documenting will be better?

Also i will update README. I propose to include in readme the installation guide but quicker ; then explain about the "h" key to get help , then explain about to view info. (Once the-old is a melpa package the user just has to call info and check for the-old)

btw, since i do not have your mail, did you see the recent reddit about an emacs rss client? i am not interested in it but you might want to look at its code. https://adamsimpson.net/writing/introducing-ivy-feedwrangler

**Comment by jerry40** [comment]
I've fixed authors macros, my mail is epanfilov at google mail. Let me think about filtering improvements. I'm thinking about possible filter which opens every article, performs search there and shows only titles which article does include search string. I'm not sure it is very useful filter though.

**Comment by pyluyten** [comment]
>  I'm thinking about possible filter which opens every article, performs search there and shows only titles which article does include search string. I'm not sure it is very useful filter though.

So what is the difference with filter 4 ? filter 4 searches in both title/content, while new one would only search content? yes if so  i do not think it is so useful

**Comment by jerry40** [comment]
4 searches in title and in short content. Full content is not downloaded, so 5 can download it and search in it. Or to google in order to ask if this article url contains the search string.

**Comment by jerry40** [comment]
Also I think to add filters by days count. Example: show articles that was published D days ago and later.
Next point is to use "client side" filters (not all, only supported by the old reader api) as "server side" to reduce trafic.
